test/src/unit-cppapi-config.cc

```diff
@@ -135,6 +135,29 @@ TEST_CASE("C++ API: Config Equality", "[cppapi][config]") {
   CHECK(config_not_equal);
 }
 
+TEST_CASE("C++ API: Optional Config", "[cppapi][config][optional]") {
+  tiledb::Config config;
+  const std::string filename = "test_config_filename";
+  config.save_to_file(filename);
+
+  const char* filenames[2] = {nullptr, filename.c_str()};
+  for (auto filename : filenames) {
```

Comment:
Prefer `GENERATE` over a loop internal to the test. `GENERATE` reports two tests run rather than one, and isolates failure reporting better.